src/vaadin-dialog.html

```diff
@@ -289,6 +340,47 @@
             e.preventDefault();
           }
         }
+
+        _setBounds(bounds) {
+          const overlay = this.$.overlay.$.overlay;
+
+          if (overlay.style.position !== 'absolute') {
+            overlay.style.position = 'absolute';
+          }
+
+          for (const arg in bounds) {
+            if (typeof bounds[arg] === 'number') {
+              bounds[arg] = `${bounds[arg]}px`;
+            }
+          }
+
+          Object.assign(overlay.style, bounds);
+        }
+
+        _snapToPixels() {
+          const overlay = this.$.overlay.$.overlay;
+          if (overlay.style.position !== 'absolute') {
+            overlay.style.maxWidth = 'auto';
```

Comment:
What is this needed for? Disabling this line doesn't seem to change the behavior in any way and there's no test for a use-case that explains it either.

If this is removed btw, `_snapToPixels()` becomes just an invocation to `_setBounds()` which makes you wonder why it needs to exist. Even now, the `overlay.style.maxWidth = 'auto'` could be inside `_setBounds` so there wouldn't need to be a separate `_snapToPixels()` method, wdyt?

Reply:
You're right, after all the refactors this method now doesn't seem to be necessary. I removed it and changed setBounds to remove the max-width. To snap to pixels all you need to do is `this._setBounds(this._getOverlayBounds())`.